rgrow/src/models/sdc1d.rs

```diff
@@ -330,6 +336,107 @@ impl SDC {
         self.kf * Molar::u0_times(bond_energy.exp())
     }
 
+    fn inverse_glue_id(g: Glue) -> Glue {
+        match g {
+            0 => 0,
+            x if x % 2 == 0 => x - 1,
+            x => x + 1
+        }
+    }
+
+    /// The fluorophore attaches to the left of the reporter
+    fn fluorophore_det_rate(&self) -> PerSecond {
+        if self.reporter_id.is_none() { return PerSecond::zero(); }
+        let fluo_glue = self.glues[(self.reporter_id.unwrap() as usize, WEST_GLUE_INDEX)];
+        let inv_glue = Self::inverse_glue_id(fluo_glue);
+        let glue_value = self.delta_g_matrix[(inv_glue, fluo_glue)]
+            - (self.temperature - Celsius(37.0)).to_celsius()
+            * self.entropy_matrix[(inv_glue, fluo_glue)];
+        let bond_energy = glue_value.times_beta(self.temperature);
+        // TODO: Is there a minus missing here ?
+        self.kf * Molar::u0_times(bond_energy.exp())
+    }
+
+    fn fluorophore_att_rate(&self) -> PerSecond {
+        self.kf * self.fluorophore_concentration
+    }
+
+    fn quencher_det_rate(&self) -> PerSecond {
+        if self.quencher_id.is_none() { return PerSecond::zero(); }
+        let quench_glue = self.glues[(self.quencher_id.unwrap() as usize, EAST_GLUE_INDEX)];
+        let inv_glue = Self::inverse_glue_id(quench_glue);
+        let glue_value = self.delta_g_matrix[(quench_glue, inv_glue)]
+            - (self.temperature - Celsius(37.0)).to_celsius()
+            * self.entropy_matrix[(quench_glue, inv_glue)];
+        let bond_energy = glue_value.times_beta(self.temperature);
+        // TODO: Is there a minus missing here?
+        self.kf * Molar::u0_times(bond_energy.exp())
+    }
+
+    fn quencher_att_rate(&self) -> PerSecond {
+        self.kf * self.quencher_concentration
+    }
+
+    pub fn monomer_change_rate_at_point<S: State>(&self, state: &S, scaffold_point: PointSafe2) -> PerSecond {
+        let strand = state.tile_at_point(scaffold_point);
+        match Some(strand) {
+            // The quencher can attach to the strand
+            q if q == self.quencher_id => self.quencher_att_rate(),
+            // The fluorophore can attach to the strand
+            r if r == self.reporter_id => self.fluorophore_att_rate(),
+            // The quencher can detach from the strand
+            Some(1) => self.quencher_det_rate(),
+            // The fluorophore can detach from the strand
+            Some(2) => self.fluorophore_det_rate(),
+            _ => PerSecond::zero(),
+        }
+    }
+
+    pub fn choose_monomer_change_at_point<S: State>(
```

Comment:
I wonder if it might be better to use tiles_max+1 and tiles_max+2 instead of 1 and 2 for the attached versions, so that then the numbering turns out a bit more intuitive (ie, the first tile listed has number 1), especially for users who might not be including reporting.